2_API_specification/Suggestions/SteffenBrinckmann/README.md

```diff
@@ -0,0 +1,67 @@
+# Light-weight Python-based No-Library API
+This is a "API" that I adopted into after 4 iterations. It is meant as a suggestion and starting point for a possible discussion.
+
+## General
+### Advantages
+- extraction does not require custom libraries
+- can be used in command-lines and dynamically imported into python (see below for code)
+- validators are easy: diff file head and tail and test if given example is extracted
+- (additional ...)
+
+### Disadvantages
+- has some python baggage around it, compared to straight conversion
+- uses an unconventional approach
+- Other languages (R, etc.) required either translation or a wrapper. I have written a CWL wrapper.
+- (additional ...)
+
+
+## Implementation
+### Extractor arguments
+- file path (could be file object to allow for data stream)
+- recipe (allows to customize extraction)
+- save fig (allows to save a "thumbnail" to file)
+- additional keywords
+
+### Extractor returns a dictionary of
+- image: a meaningfull thumbnail: base64-encoded or svg
+- metaVendor: metadata directly copied from vendor file
+- metaUser: metadata the scientist creates in the extractor
```

Comment:
should one rely here on one of the widely used metadata schemes (something like Dublin Core)?

Reply:
That is a very good and important point and going into the typical competition of interoperability vs. low-entry-barrier.
- free schemas or literal translations of what the instrument vendor uses are easy and people can easily write/read them. 
- fixed schema are required for interoperability but require often extensive study, one has to read and understand. Plus, sometimes the established schema are not 100% fitting. Dublin core is great of research and treats the extractor as a piece of scientific-work (which it is), schema.org is more general.

I, personally, right now stick to the literal translation of the scientific instrument vendor plus add some add hoc terms that scientists *might* understand (areaUnderCurve). But I am fully aware that the interoperability is not there and looking forward to the bigger discussion of the competition interoperability vs. entry-barrier.

Reply:
So far in https://github.com/marda-alliance/metadata_extractors_schema I have included a few mappings to Dublin Core and Schema.org (via LinkML's `slots_uri` field). I don't know how useful it is to have a smattering of these around, so perhaps we could find a use case for these mappings amongst ourselves